### ISSUE TITLE
Initialize Date() in ISO-8601 format

### ISSUE THREAD

**Issue by zanonmark** [vaadin/vaadin-date-picker#792]
### Description
When (Enhanced)DatePicker is set to use "dd/MM/yyyy" as pattern and "dd/MM/yyyy", "dd/MM/yy" as parsers, inputting a year before 1000 will result in a wrong date being evaluated at client-side.

i.e.: 13/01/0040 or 13/01/40 (13th January 40, not 1940 or 2040)
-> server: 0040-01-13 (ok)
-> client: 1940-01-13 (wrong)

### Expected outcome
Both server and client should parse the string properly as 0040-01-13.

Maybe at some point the JavaScript Date() object is initialized in Date(year, month, day) format and not in Date(iso8601String) format?

Thanks,
MZ

**Comment by zanonmark** [comment]
I was reading this [CookBook recipe](https://cookbook.vaadin.com/date-picker-pattern), which is not directly related to vaadin-date-picker but the issue may be the same: instead of initializing the Date() object as a (year, month, day) triple, shouldn't it use an ISO-8601-formatted date string?

Thanks,
MZ

**Comment by tomivirkki** [comment]
If the issue is about how the Enhanced Date Picker parses the date string, and not about vaadin-date-picker specifically, please report the issue in https://github.com/vaadin-component-factory/enhanced-date-picker

**Comment by zanonmark** [comment]
No, it's also for DatePicker: try [https://vaadin.com/components/vaadin-date-picker/java-examples](https://vaadin.com/components/vaadin-date-picker/java-examples), pick the first example and enter today's date 3/17/0084 (US locale), it will become 3/17/1984.

EnhancedDatePicker just inherits this issue.

Thanks,
MZ
